Dockerfile

```diff
@@ -59,3 +67,32 @@ RUN touch /var/log/stats.log /var/log/dump_create.log && chown listenbrainz:list
 # Make sure the cron service doesn't start automagically
 # http://smarden.org/runit/runsv.8.html
 RUN touch /etc/service/cron/down
+
+# Consul Template service is already set up with the base image.
+# Just need to copy the configuration.
+COPY ./docker/consul-template.conf /etc/consul-template.conf
+COPY ./docker/$deploy_env/uwsgi/uwsgi.service /etc/service/uwsgi/run
+RUN chmod 755 /etc/service/uwsgi/run
+COPY ./docker/$deploy_env/uwsgi/uwsgi.ini /etc/uwsgi/uwsgi.ini
+COPY ./docker/prod/uwsgi/uwsgi-api-compat.ini /etc/uwsgi/uwsgi-api-compat.ini
+
+# Create directories for backups and FTP syncs
+RUN mkdir /home/listenbrainz/backup /home/listenbrainz/ftp
+RUN chown -R listenbrainz:listenbrainz /home/listenbrainz/backup /home/listenbrainz/ftp
+
+# Now install our code, which may change frequently
+COPY . /code/listenbrainz/
+
+RUN mkdir /static
+WORKDIR /static
+
+# Compile static files
+RUN curl -sL https://deb.nodesource.com/setup_10.x | bash - && \
```

Comment:
not a good idea to run apt after copying code. If this is something that only happens in this image then it should be done as the first step.
If you don't want node_modules in production, you could do npm install in the base image, and then in the prod image just copy node_modules from the base image, run npm, and then delete it (I'm not sure with x thousand small files if this will be efficient or not). It'll prevent us from having to download from npm on every deploy, but it'd require adding package.json to the base image.

Reply:
This was a mistake, it now copies the static files and installs and then copies code.